pyaerocom/io/readungridded.py

```diff
@@ -827,7 +827,12 @@ def read(
             if data is None:
                 data = data_to_append
             else:
-                data.append(data_to_append)
+                try:
+                    data.append(data_to_append)
+                except ValueError:
+                    # UngriddedData cannot add UngriddedDataStructured, while the other way works
+                    # just re-ordering here in such cases
+                    data = data_to_append.append(data)
```

Comment:
[nitpick] I realize that this may be out of scope but considering other append functions in Python modify the object in place and return None (see eg. list.append) I think this code is strange to read. I think this change makes it easier to read, though ideally .append should be modified to be more similar to other append functions.
```suggestion
                    data_to_append.append(data)
                    data = data_to_append
```